Ginger/Ginger/ApplicationModelsLib/POMModels/POMsSelectionPage.xaml.cs

```diff
@@ -0,0 +1,164 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+
+namespace Ginger.ApplicationModelsLib.POMModels
+{
+    /// <summary>
+    /// Interaction logic for POMsSelectionPage.xaml
+    /// </summary>
+    public partial class POMsSelectionPage : Page
+    {
+        public ObservableList<POMBindingObjectHelper> PomModels = new ObservableList<POMBindingObjectHelper>();
+        SingleItemTreeViewSelectionPage mApplicationPOMSelectionPage = null;
+
+        public delegate void POMSelectionEventHandler(object sender, Guid guid);
+
+        public event POMSelectionEventHandler POMSelectionEvent;
+
+        public bool ShowTitle { get; set; }
+
+        /// <summary>
+        /// This event is used to raise object
+        /// </summary>
+        public void POMSelectedEvent(Guid guid)
+        {
+            if (POMSelectionEvent != null)
+            {
+                POMSelectionEvent?.Invoke(this, guid);
+            }
+        }
+        
+        /// <summary>
+        /// Ctor for default settings
+        /// </summary>
+        public POMsSelectionPage()
+        {
+            InitializeComponent();
+            SetPOMGridView();            
+        }
+
+        /// <summary>
+        /// This method is used to set the columns POM GridView
+        /// </summary>
+        private void SetPOMGridView()
+        {
+            xGridPOMListItems.SetTitleLightStyle = true;
+            GridViewDef view = new GridViewDef(GridViewDef.DefaultViewName);
+            view.GridColsView = new ObservableList<GridColView>();
+            view.GridColsView.Add(new GridColView() { Field = nameof(POMBindingObjectHelper.ContainingFolder), Header = "Folder", WidthWeight = 100, AllowSorting = true, BindingMode = BindingMode.OneWay, ReadOnly = true });
+            view.GridColsView.Add(new GridColView() { Field = nameof(POMBindingObjectHelper.ItemName), Header = "Name", WidthWeight = 150, AllowSorting = true, BindingMode = BindingMode.OneWay, ReadOnly = true });
+            xGridPOMListItems.btnAdd.Click -= BtnAdd_Click;
+            xGridPOMListItems.btnAdd.Click += BtnAdd_Click;
+            xGridPOMListItems.SetAllColumnsDefaultView(view);
+            xGridPOMListItems.InitViewItems();
+            PomModels = new ObservableList<POMBindingObjectHelper>();
+            xGridPOMListItems.DataSourceList = PomModels;
+            xGridPOMListItems.Title = "Selected POM's";
+            xGridPOMListItems.ShowTitle = ShowTitle ? Visibility.Visible : Visibility.Collapsed;
+        }
+
+        /// <summary>
+        /// This event is used to open the popup for POM selection
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void BtnAdd_Click(object sender, RoutedEventArgs e)
+        {
+            if (mApplicationPOMSelectionPage == null)
+            {
```

Comment:
need to consider that due to creation of page only if it null then in case any change happend to POM tree it won't be reflected. Prateek added generic Refresh capability to the TreeSelectionPage, make sure you using it.

Reply:
Added code